### ISSUE TITLE
A Linear layer may be reused mistakely.

### ISSUE THREAD

**Issue by youngornever** [LooperXX/DF-Net#2]
You use the same Linear layer in https://github.com/LooperXX/DF-Net/blob/master/models/modules.py:227;
"""
        outputs_ = self.W(outputs)
        hidden_ = self.W(hidden)
"""
Though It'is ok, I guess this may be reused mistakely.
I guess the second layer is self.W_hid = nn.Linear(2 * hidden_size, hidden_size); see the 194 line in the module.py

**Comment by LooperXX** [comment]
Hi @youngornever:
    Sorry to make you confused. 
    The 194-line in the module.py should be deleted although this does not affect the results of the experiment.
    It was the early experimental code of our model and we found that output and hidden using the same linear layer has better and stable performance. Since the function of this linear layer is to reduce the dimension so that the dimension could match. We discard the W_hid and use the same W in the official version.

**Comment by youngornever** [comment]
ok. Thanks for your reply.
Another question：
See the 165-line, why do you softmax over the seqlen (i.e., dim=1), but sum over the domain?
It is very strange. Usually we softmax over a dim, then weighted_sum over this dim.

**Comment by youngornever** [comment]
@LooperXX 
ok. Thanks for your reply.
Another question：
See the 165-line, why do you softmax over the seqlen (i.e., dim=1), but sum over the domain?
It is very strange. Usually we softmax over a dim, then weighted_sum over this dim.

**Comment by LooperXX** [comment]
Hi @youngornever:
Thank you for pointing out this question. We are sorry to make this mistake carelessly. 
We fix this question and rerun the experiments on the SMD and MultiWOZ. The results are better than the current version. 
You can reload the new best models to check the following results.

Datasets | BLEU | F1 | Navigate F1 | Weather F1 | Calendar F1
-- | -- | -- | -- | -- | --
SMD | 15.22 | 63.06 | 57.63 | 57.25 | 74.57
MultiWOZ| 9.74 | 35.7 | 42.15 | 32.84 | 30.74

Thanks for your kindful mention again!
If you have any further questions, you can directly contact me via LooperXX@outlook.com to get a quick reply.
